### ISSUE TITLE
CodFreq allow entry point from .sam file

### ISSUE THREAD

**Issue by TKMarkCheng** [hivdb/codfreq#10]
Thank you for this wonderful pipeline.

Is it possible to add in the functionality to allow for entering in processed .bam file (adapter trimmed, aligned, primer trimmed) instead of fastqs?

I wanted to experiment with different pre-processing methods.

**Comment by gpcr** [comment]
python3 /path_to_codfreq/scripts>/sam2codfreq.py \
    -r /path_to_your_profile>/HIV1.json \
    -o /path_to_output_codfreq_file_name..json \
    /path_to_HIV1_aligned_bam_file.bam

**Comment by TKMarkCheng** [comment]
I initially thought the same, but [codfreq/**scripts**/sam2codfreq.py](https://github.com/hivdb/codfreq/blob/main/scripts/sam2codfreq.py) does not allow the input of a profile.

> `usage: codfreq/scripts/sam2codfreq.py <SAMFILE> <OUTPUT>`

I've noticed [codfreq/**codfreq**/sam2codfreq.py](https://github.com/hivdb/codfreq/blob/main/codfreq/sam2codfreq.py) does but is more complex and is not a terminal callable script (as it does not have)
```python
if __name__ == '__main__':
  main()
```

**Comment by philiptzou** [comment]
Yeah it’s currently not an exposed feature. However I believe you can
create a main script, then import and expose the function from sam2codfreq
to achieve what you want.
